README.md

```diff
@@ -22,13 +61,65 @@ Transaction are applied atomically. Queries can't retrieve partially committed d
 ### Durability
 Database should be restored to consistent state if IO is interrupted at any point. 
 
+#### Review
+
+I firstly misunderstood 'Durability' (IO failing not breaking db) for some consitency (call to commit means data is written in db).
+So there is many irrelevant notes in code.
+Call to commit does not actualy mean data is written/flushed to disk.
+- commit queue in db.rs only write to in memory queue.
+- mmap not having flush guarantee, this is something I am not really confident in.
+- flush worker also defers log flushing to file.
+
+So the db requires graceful shutdown.
+If it does not happen, state can be incorrect.
+
+
+About 'commit' semantic, this is a bit more awkward.
+All should be fine as long as all persistence is in a single db, but we should be strongly underline
+that commit does not flush to disk.
+So if process is sigkilled, the state would be consistent (but if another persistence source is used, it could be
+in a different state than this db).
+
+If full system is kill, mmap could not have flush: is it problematic?
+
+-> using a single db (no rocksdb in pkdot) seems rather prioritary.
+(IIUC, current commit semantic behave like rocksdb with 'DBOptions::manual_wal_flush')
+(which can be a gain for rocksdb too (but would require some periodic similar flush call).
```

Comment:
Yeah, the fact that indexes and value tables are not msync/fsynced after enacting the log (if sync mode is enabled) is a big overlook on my part. I'll add a separate issue for that.
So currently, as you've noted, even if sync mode is not enabled, there's still consistency guarantee at the process level. I.e. graceful shutdown of the database process is not really required. It may be interrupted at any moment, and as long as OS propagates all the pending writes, database will be recovered with the log. However if the OS/power is shut down unexpectedly with some of the IO performed out of order, this may lead to issues indeed.